python/dgl/heterograph.py

```diff
@@ -2052,9 +2053,11 @@ def node_type_subgraph(self, ntypes):
                 edge_frames.append(self._edge_frames[i])
 
         metagraph = graph_index.from_edge_list(meta_edges, True)
+        # num_nodes_per_type doesn't need to be int32
         hgidx = heterograph_index.create_heterograph_from_relations(
-            metagraph, rel_graphs, utils.toindex(num_nodes_per_type))
-        hg = DGLHeteroGraph(hgidx, ntypes, induced_etypes, node_frames, edge_frames)
+            metagraph, rel_graphs, utils.toindex(num_nodes_per_type, "int64"))
```

Comment:
Why not make the dtype consistent with `rel_graphs`?

Reply:
Currently data related to metagraph are all int64. And the C++ `HeteroGraph` uses `std::vector<int64_t> num_verts_per_type_`. This is mainly for simplicity, otherwise lots of extra changes may needed.

Reply:
I see. You could put that in the C++ header docstring.